### PR TITLE
new module jvarkit/dict2bed

### PR DESCRIPTION
**Pull request by lindenb**
this is a PR for jvarkit/vcf2table  which convert files containing a dictionary like cram/bam/dict/etc.. to bed: https://jvarkit.readthedocs.io/en/latest/DictToBed/

```
$ java -jar dist/jvarkit.jar dict2bed ~/src/jvarkit-git/src/test/resources/*.bam |  head | column -t
chrom  start  end        path                                                                         tid  buildName  AS  M5  SP  UR
chr1   0      248956422  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  0    GRCh38     .   .   .   .
chr2   0      242193529  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  1    GRCh38     .   .   .   .
chr3   0      198295559  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  2    GRCh38     .   .   .   .
chr4   0      190214555  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  3    GRCh38     .   .   .   .
chr5   0      181538259  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  4    GRCh38     .   .   .   .
chr6   0      170805979  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  5    GRCh38     .   .   .   .
chr7   0      159345973  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  6    GRCh38     .   .   .   .
chr8   0      145138636  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  7    GRCh38     .   .   .   .
chr9   0      138394717  /home/lindenb/src/jvarkit-git/src/test/resources/ENCFF331CGL.rnaseq.b38.bam  8    GRCh38     .   .   .   .
```

## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [X] `nf-core modules test <MODULE> --profile docker`
    - [X] `nf-core modules test <MODULE> --profile singularity`
    - [X] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/jvarkit/dict2bed/main.nf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    output:
+    tuple val(meta), path(dict_files), path("*.bed"), emit: bed
+    path "versions.yml"                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args1          = task.ext.args1 ?: ''
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/dict2bed/main.nf:21]: I think it's just `args`

File: modules/nf-core/jvarkit/dict2bed/main.nf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/jvarkit:2024.08.25--hdfd78af_1':
+        'biocontainers/jvarkit:2024.08.25--hdfd78af_1' }"
+
+    input:
+    tuple val(meta),  path(dict_files)
+
+    output:
+    tuple val(meta), path(dict_files), path("*.bed"), emit: bed
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/dict2bed/main.nf:14]:
```suggestion
    tuple val(meta), path("*.bed"), emit: bed
```

I'd rather we just emit the output, and not re-emit the input

**Review by maxulysse — COMMENTED** [review]
Looking good, just a tiny amount of polish

File: modules/nf-core/jvarkit/dict2bed/meta.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  - fai
+tools:
+  - "jvarkit":
+      description: "Java utilities for Bioinformatics."
+      homepage: "https://github.com/lindenb/jvarkit"
+      documentation: "https://jvarkit.readthedocs.io/"
+      tool_dev_url: "https://github.com/lindenb/jvarkit"
+      doi: "10.6084/m9.figshare.1425030"
+      licence: ["MIT License"]
+      args_id: "$args"
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/dict2bed/meta.yml:20]: any idea if we need to refer args2?

Comment (lindenb) [reply modules/nf-core/jvarkit/dict2bed/meta.yml:20]: no-idea, you're the boss of nf-core here :-)

**Review by nvnieuwk — APPROVED** [review]
LGTM

**Pull request merged**